Added Transform Functionality to DataTable.svelte & updated README.md

anti-h3r0 opened the pull request:
## Description
title says it all

Closes #299 .

anti-h3r0 committed:
Added Transform Functionality to DataTable.svelte & updated README.md

diegofhe commented on src/components/DataTable.svelte:
```diff
@@ -190,6 +190,31 @@
+			const tempRow = cloneDeep(row);
+			let val;
+			if (typeof transform === 'function') {
+				val = get(tempRow, field, null);
```
null can be omitted, is the default

diegofhe commented on src/components/DataTable.svelte:
```diff
@@ -190,6 +190,31 @@
+				if (val) set(tempRow, field, transform(val));
+			} else if (typeof transform === 'object' && !Array.isArray(transform)) {
+				const target = transform.field || field;
+				val = get(tempRow, target, null);
```
same here

diegofhe commented on src/components/DataTable.svelte:
```diff
@@ -190,6 +190,31 @@
+				}
+			}
+			else {
+				throw new Error('transform is not configured correctly, please read the documentation')
```
nice

diegofhe reviewed (approved):
LGTM, Just see my comments

diegofhe commented on src/components/DataTable.svelte:
```diff
@@ -190,6 +190,31 @@
+			else {
+				throw new Error('transform is not configured correctly, please read the documentation')
+            }
+			if (!val) console.error(`the property 'row.${transform.field || field}' is undefined.\n\t\t assigning value to null`)
```
Watch out. This will console on falsy values like 0 and false. I would check for undefined or null for this kind of logic. But for this specific case, I would delete this console error, maybe the user wants the property to be undefined or null

anti-h3r0 replied:
oh okay

anti-h3r0 committed:
renamed transform fields and added culling option to cull all data except for the transformed data.

experimented with scope & side-effects

updated the README.md & added an example which explores the transform scope and side-effects

anti-h3r0 committed:
removed redundant defaults and deleted console.error when target value is falsy

anti-h3r0 committed:
remove console.log in App.svelte

anti-h3r0 committed:
fix/transform -- bug fixes, edited some behavior, and updated the README.md

anti-h3r0 committed:
update README.md note behavior when property doesn't exist.

diegofhe reviewed (approved):
LGTM
